python/pyspark/sql/types.py

```diff
@@ -1555,6 +1555,26 @@ def treeString(self, maxDepth: int = JVM_INT_MAX) -> str:
             field._build_formatted_string(prefix, stringConcat, depth)
         return stringConcat.toString()
 
```

Comment:
Would need to mark it as `@classmethod`. One thing I am not sure is that we should have the running Spark session before run this API ... otherwise, it will automatically start a context ... one way is to `getActiveSession`, and throw an exception if any session does not exist.

Reply:
yes, we should avoid create a new context.

Reply:
but why should be a `@classmethod`

Reply:
oh no. it doesn't have to be.